com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/DoFGather.compute

```diff
@@ -88,106 +118,293 @@ float2 PointInCircle(float angle)
     return float2(cos(angle), sin(angle)) * float2 (1 - Anamorphism, 1 + Anamorphism);
 }
 
-[numthreads(GROUP_RES, GROUP_RES, 1)]
-void KMain(uint3 dispatchThreadId : SV_DispatchThreadID)
+void ResolveColorAndAlpha(inout float4 outColor, inout float outAlpha, CTYPE defaultValue)
 {
-    PositionInputs posInputs = GetPositionInput(float2(dispatchThreadId.xy), _ScreenSize.zw, uint2(GROUP_RES, GROUP_RES));
-
-    // Bucket 0 : far focus region
-    // Bucket 1 : in focus region
-    // Bucket 2 : near focus region
-    float4 totalColor[NUM_BUCKETS];
-    float prevRingWeight[NUM_BUCKETS];
-    float totalHits[NUM_BUCKETS];
+    outColor.xyz = outColor.w > 0 ? outColor.xyz / outColor.w : defaultValue.xyz;
 #ifdef ENABLE_ALPHA
-    float totalAlpha[NUM_BUCKETS];
+    outAlpha = outColor.w > 0 ? outAlpha / outColor.w : defaultValue.w;
 #endif
+}
 
-    // Initialize color/weights and everything else to zero (for all buckets)
-    for (int i = 0; i < NUM_BUCKETS; ++i)
+#ifdef FAST_APPROXIMAION
+
+void AccumulateSample(SampleData sampleData, float centerCoC, float2 limit, inout float4 colorAccum, inout float alphaAccum)
+{
+    float dw = 1;
+    if (sampleData.CoC >= 0 && abs(centerCoC) < limit.y)
     {
-        totalColor[i] = 0.0f;
-        prevRingWeight[i] = 0.0f;
-#ifdef ENABLE_ALPHA
-        totalAlpha[i] = 0.0f;
-#endif
+        // occlusion test: if the central CoC occludes the sample that we gather, then the sample weight should be zero
+        dw = (sampleData.CoC <= limit.x * centerCoC) ? 1.0 : 0.0;
     }
 
-    CTYPE originalColor = LOAD_TEXTURE2D_X(_InputTexture, posInputs.positionSS).CTYPE_SWIZZLE;
+    float weight = dw * GetSampleWeight(abs(sampleData.CoC));
+    colorAccum += float4(sampleData.color.xyz * weight, weight);
+#ifdef ENABLE_ALPHA
+    alphaAccum += sampleData.color.w * weight;
+#endif
+}
 
-    // Note: for the far-field, we don't need to search further than than the central CoC.
-    // If there is a larger CoC that overlaps the central pixel then it will have greater depth
-    float centerCoC = GetCoCRadius(posInputs.positionSS);
-    float maxRadius = abs(centerCoC);
+void DoFGatherFast(PositionInputs posInputs, int numSamples, float maxRadius, SampleData centerSample, out float4 outColor, out float outAlpha)
+{
+    AccumData accumData;
+    InitAccumData(accumData);
```

Comment:
Can use ZERO_INITIALIZE like we do in most places to 0-init structs, but this is fine as well (mentioning if you want to keep code leaner)